src/hooks/useExpression.js

```diff
@@ -0,0 +1,59 @@
+import { useState } from 'react';
+
+import { arithmetic } from '../utils/arithmetic';
+
+import { INPUT_NUMBER_FIRST_MESSAGE, NUMBER_OF_EXCEPTION_MESSAGE } from '../constants/error';
+
+const useExpression = () => {
+  const [expression, setExpression] = useState({ number: [], operation: [], counter: 0 });
+
+  const writeExpression = (buttonValue, totalText) => {
+    const { number, operation } = expression;
+    let counter = expression.counter;
+
+    if (isNaN(buttonValue) && !number[counter]) {
+      alert(INPUT_NUMBER_FIRST_MESSAGE);
+      return totalText;
+    }
+
+    if (isNaN(buttonValue)) {
+      operation.push(buttonValue);
+      counter = counter + 1;
+    }
+
+    if (number[counter] && number[counter].length >= 3) {
+      alert(NUMBER_OF_EXCEPTION_MESSAGE);
+      return '';
+    }
+
+    if (!isNaN(buttonValue)) {
+      number[counter] = number[counter] ? number[counter] + buttonValue : buttonValue;
+    }
+
+    setExpression({ number, operation, counter });
+    return buttonValue;
+  };
+
+  const calculateExpression = () => {
+    const number = expression.number.map(Number);
+    const operation = expression.operation[0];
+
+    const result = arithmetic(number[0], number[1], operation);
+
+    return ResetExpression(result);
+  };
+
+  const ResetExpression = (result = '') => {
+    if (result === '') {
+      setExpression({ number: [], operation: [], counter: 0 });
+      return 0;
+    }
+
+    setExpression({ number: [result], operation: [], counter: 0 });
```

Comment:
if 문을 줄이려면 표현식을 많이 써야 합니다. 😃 

```suggestion
    const number = result === "" ? : [] : [result];
    setExpression({ number, operation: [], counter: 0 });
```

Reply:
항상 유의해서 작성하도록 하겠습니다!